File: SoClover.Server/Models/Requests/Requests.cs
namespace SoClover.Server.Models.Requests
{
    public class CreateRoomRequest
    {
        public string PlayerName { get; set; } = string.Empty;
    }

    public class JoinRoomRequest
    {
        public string RoomCode { get; set; } = string.Empty;
        public string PlayerName { get; set; } = string.Empty;
    }


    public class SubmitCluesRequest
    {
        public string[] Words { get; set; } = [];
    }

    public class RotateCardRequest
    {
        public int GameRoomCardId { get; set; }
    }

    public class MoveCardRequest
    {
        public int GameRoomCardId { get; set; }
        public int PositionIndex { get; set; }
    }
}

File: soclover.client/src/models/requests.ts
export interface CreateRoomRequest {
    playerName: string;
    playerGuid: string;
}

export interface JoinRoomRequest {
    roomCode: string;
    playerName: string;
    playerGuid: string;
}

export interface SubmitCluesRequest {
    words: string[];
}

export interface RotateCardRequest {
    gameRoomCardId: number;
}

export interface MoveCardRequest {
    gameRoomCardId: number;
    positionIndex: number;
}
